### ISSUE TITLE
Search of hotel only by name has not optimal rankinng

### ISSUE THREAD

**Issue by dbf256** [organicmaps/organicmaps#4119]
Hotel is stored as "Hotel Don Paco" is OSM.
If I try to search it using only name "Don Paco" it is ranked outside of the first pageneven if I am closed to it and some POI that starts with "Don" only are displayed higher. Probably it is because they starts with this string, not just contains it. 
![Screenshot_2022-12-20-09-19-47-241_app organicmaps](https://user-images.githubusercontent.com/2550295/208618752-a48b21d8-a7c1-4f6a-8323-eb8c174de628.png)
![Screenshot_2022-12-20-09-20-16-833_app organicmaps](https://user-images.githubusercontent.com/2550295/208618761-2e29e7f5-92ed-4a20-9d12-f222a32264ba.png)
![Screenshot_2022-12-20-09-19-59-849_app organicmaps](https://user-images.githubusercontent.com/2550295/208618764-cebc8145-8921-4499-ae60-e0bfef378664.png)

**Comment by biodranik** [comment]
Is it a correct name? Features usually don't have their type in the name.

**Comment by dbf256** [comment]
> Is it a correct name? Features usually don't have their type in the name.

I'm not 100% sure about osm rules (also Name field is not simple from this point of view) but it is a common case for different regions.

![Screenshot_2022-12-20-12-26-07-434_app organicmaps](https://user-images.githubusercontent.com/2550295/208658527-117b725d-120c-4b21-a37c-af6f65e5b383.png)
![Screenshot_2022-12-20-12-33-09-987_app organicmaps](https://user-images.githubusercontent.com/2550295/208658530-13e2716b-6ff0-4236-b153-fb7d6cede8ca.png)
![Screenshot_2022-12-20-12-37-23-657_app organicmaps](https://user-images.githubusercontent.com/2550295/208658534-5e418385-af7b-4c2e-bb82-569df94941cc.png)

**Comment by vng** [comment]
Yes, ranking is strange here, because first place for "Don Paco" query (no space at the end) is:
"Don Luis" apartments somewhere near "Av los Pacos" street. We should definitely reduce rank for POIs _nearby_ Streets if there is not exact address matching.

**Comment by vng** [comment]
Also good example is "San Francisco" search within Buenos Aires. Top "San XXX" results near some "YYY Francisco" streets.

**Comment by biodranik** [comment]
> Also good example is "San Francisco" search within Buenos Aires. Top "San XXX" results near some "YYY Francisco" streets.

That should be easy to fix by searching the world index first for exact matches and prioritizing them.
